contentcuration/contentcuration/frontend/shared/data/serverSync.js

```diff
@@ -80,6 +81,116 @@ function trimChangeForSync(change) {
+      .anyOf(disallowedRevs)
+      .modify({ disallowed: true, synced: true });
+  }
+  return Promise.resolve();
+}
+
+function socketHandleReturnedChanges(returnedChange) {
+  // The changes property is an array of any changes from the server to apply in the
+  // client.
+  let returnedChanges = [];
+  returnedChanges.push(returnedChange);
+  if (returnedChanges.length) {
+    return applyChanges(returnedChanges);
+  }
+  return Promise.resolve();
+}
+
+function socketHandleErrors(error) {
+  // The errors property is an array of any changes that were sent to the server,
+  // that were rejected, with an additional errors property that describes
+  // the error.
+  let errors = [];
+  errors.push(error);
+  if (errors.length) {
+    const errorMap = {};
+    for (let error of errors) {
+      errorMap[error.server_rev] = error;
+    }
+    // Set the return error data onto the changes - this will update the change
+    // both with any errors and the results of any merging that happened prior
+    // to the sync operation being called
+    return db[CHANGES_TABLE].where('server_rev')
+      .anyOf(Object.keys(errorMap))
+      .modify(obj => {
+        return Object.assign(obj, errorMap[obj.server_rev]);
+      });
+  }
+  return Promise.resolve();
+}
+
+function socketHandleSuccesses(success) {
+  // The successes property is an array of server_revs for any previously synced changes
+  // that have now been successfully applied on the server.
+  if (success.length) {
+    return db[CHANGES_TABLE].where('server_rev')
+      .anyOf(success.server_rev)
+      .delete();
+  }
+  return Promise.resolve();
+}
+
+function socketHandleMaxRevs(change, userId) {
+  let changes = [];
+  changes.push(change);
+  const allChanges = orderBy(changes, 'server_rev', 'desc');
+  const channelIds = uniq(allChanges.channel_id).filter(Boolean);
+  const maxRevs = {};
+  const promises = [];
+  for (let channelId of channelIds) {
+    const channelChanges = allChanges.filter(c => c.channel_id === channelId);
+    maxRevs[`${MAX_REV_KEY}.${channelId}`] = channelChanges[0].server_rev;
+    const lastChannelEditIndex = findLastIndex(
+      channelChanges,
+      c => !c.errors && !c.user_id && c.created_by_id && c.type !== CHANGE_TYPES.PUBLISHED
+    );
+    const lastPublishIndex = findLastIndex(
+      channelChanges,
+      c => !c.errors && !c.user_id && c.created_by_id && c.type === CHANGE_TYPES.PUBLISHED
+    );
+    if (lastChannelEditIndex > lastPublishIndex) {
+      promises.push(
+        Channel.transaction({ mode: 'rw', source: IGNORED_SOURCE }, () => {
+          return Channel.table.update(channelId, { unpublished_changes: true });
+        })
+      );
+    }
+  }
+  const lastUserChange = allChanges.find(c => c.user_id === userId);
+  if (lastUserChange) {
+    maxRevs.user_rev = lastUserChange.server_rev;
+  }
+  if (Object.keys(maxRevs).length) {
+    promises.push(Session.updateSession(maxRevs));
+  }
+  return Promise.all(promises);
+}
```

Comment:
I think it was mentioned that the call to this with a single `change` object means all of the code in this function can be simplified.